pypose/optim/kernel.py

```diff
@@ -123,3 +123,149 @@ def forward(self, input: Tensor) -> Tensor:
         '''
         assert torch.all(input >= 0), 'input has to be non-negative'
         return self.delta2 * (input/self.delta2 + 1).log()
+
+
+class SoftLOne(nn.Module):
+    r"""The robust SoftLOne kernel cost function.
+
+    .. math::
+        \bm{y}_i=2\left ( \delta \sqrt{\frac{1}{{\delta{}}^{2}}+\bm{x}_i}- 1\right )
+
+    where :math:`\delta` (delta) is a hyperparameter, :math:`\bm{x}` and :math:`\bm{y}` are the
+    input and output tensors, respectively.
+
+    Args:
+        delta (float, optional): Specify the SoftLOne cost. The value must be positive. Default: 1.0
+
+    Note:
+        The input has to be a non-negative tensor and the output tensor has the same shape with the
+        input.
+
+    Example:
+        >>> import pypose.optim.kernel as ppok
+        >>> kernel = ppok.SoftLOne()
+        >>> input = torch.tensor([0, 0.5, 1, 2, 3])
+        >>> kernel(input)
+        tensor([0.0000, 0.4495, 0.8284, 1.4641, 2.0000])
+    """
+    def __init__(self, delta: float = 1.0) -> None:
+        super().__init__()
+        assert delta > 0, ValueError("delta has to be positive: {}".format(delta))
+        self.delta1 = delta
+        self.delta2 = delta**2
+
+    def forward(self, input: Tensor) -> Tensor:
+        '''
+        Args:
+            input (torch.Tensor): the input tensor (non-negative).
+        '''
+        assert torch.all(input >= 0), 'input has to be non-negative'
+        return 2*(self.delta1*(1/self.delta2+input).sqrt()-1)
+
+
+class Arctan(nn.Module):
+    r"""The robust Arctan kernel cost function.
+
+    .. math::
+        \bm{y}_i=\delta ^{2}\arctan \left ( \frac{\bm{x}_i}{\delta ^{2}}\right )
+
+    where :math:`\delta` (delta) is a hyperparameter, :math:`\bm{x}` and :math:`\bm{y}` are the
+    input and output tensors, respectively.
+
+    Args:
+        delta (float, optional): Specify the Arctan cost. The value must be positive. Default: 1.0
```

Comment:
`delta` doesn't have to be positive, since you always use `delta**2`
remember to change code and doc simultaneously.